src/databricks/sql/thrift_backend.py

```diff
@@ -260,19 +265,40 @@ def make_request(self, method, request):
         def get_elapsed():
             return time.time() - t0
 
+        def make_bounded_delay(attempt: int, delay: int):
```

Comment:
Don't block on this, but IMO this `make_request` method is getting a bit unwieldy with all the various inner methods. It would be good if we could put this retry logic in a separate module/class in a follow up, WDYT?

Reply:
Agreed. I'll file a ticket for this.